rust/src/server.rs

```diff
@@ -71,11 +83,21 @@ where
                     match (split.next(), split.next()) {
                         (Some("Bearer"), Some(token)) => {
                             let tokens = server.tokens.read().await;
-                            if !tokens.contains_key(token) {
-                                return Ok(Status::unauthenticated("invalid auth token").to_http());
+                            match tokens.get(token) {
+                                Some(val) => match val.parse() {
```

Comment:
`val` makes this unclear to me, can you call it `maybe_tag` or something to show that it's the tag that you want to parse?